### PR TITLE
[RFR] Better chart

### PR DESCRIPTION
**Pull request by ThieryMichel**

### CODE REVIEW

File: src/api/statements/LodexReduceQuery.js [hunk 1]
```diff
@@ -27,8 +34,22 @@ export const createFunction = MongoClientImpl =>
         const collName = String('mp_').concat(
             hashCoerce.hash({ reducer, fields }),
         );
+
+        const handleDb = await MongoClientImpl.connect(
```
Comment (fzaninotto) [inline src/api/statements/LodexReduceQuery.js:38]: The connection is opened but never closed. This is a memory leak.

You'd better inject a connection managed outside.

File: src/api/statements/LodexRunQuery.js [hunk 2]
```diff
@@ -2,34 +2,48 @@ import { MongoClient } from 'mongodb';
...
-        const limit = this.getParam('limit', data.$limit || 10);
-        const skip = this.getParam('skip', data.$skip || 0);
-        const sort = this.getParam('sort', data.$sort || {});
+        const query = this.getParam('query', data.query || {});
+        const limit = this.getParam('limit', data.limit || 10);
+        const skip = this.getParam('skip', data.skip || 0);
+        const sort = this.getParam('sort', data.sort || {});
         const target = this.getParam('total');
 
         const handleDb = await MongoClientImpl.connect(
```
Comment (fzaninotto) [inline src/api/statements/LodexRunQuery.js:22]: Same as above, the connection should be initialized in a central place, and kept in the context/ in a singleton.

File: src/app/js/formats/distributionChart/global-bar-chart/Component.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
```
Comment (fzaninotto) [inline src/app/js/formats/distributionChart/global-bar-chart/Component.js]: rename the file to `BarChartView`

File: src/app/js/formats/distributionChart/global-bar-chart/Component.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    render() {
+        const { colorSet, chartData } = this.props;
+
+        return (
+            <ResponsiveContainer width={600} height={300}>
+                <BarChart
+                    data={chartData}
+                    layout="vertical"
+                    margin={{ top: 15, right: 50, left: 0, bottom: 0 }}
```
Comment (fzaninotto) [inline src/app/js/formats/distributionChart/global-bar-chart/Component.js]: double mustache

File: src/app/js/formats/distributionChart/global-bar-chart/Component.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                    maxBarSize={10}
+                >
+                    <XAxis type="number" dataKey="value" />
+                    <YAxis
+                        type="category"
+                        dataKey="name"
+                        interval={0}
+                        padding={{ top: 3, bottom: 3 }}
```
Comment (fzaninotto) [inline src/app/js/formats/distributionChart/global-bar-chart/Component.js]: double mustache

File: src/app/js/formats/distributionChart/global-pie-chart/Component.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
```
Comment (fzaninotto) [inline src/app/js/formats/distributionChart/global-pie-chart/Component.js]: rename to `PieChartView`

File: src/app/js/formats/distributionChart/global-radar-chart/Component.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (fzaninotto) [inline src/app/js/formats/distributionChart/global-radar-chart/Component.js]: rename to `RadarChartView`

File: src/app/js/formats/resources-grid/Component.js [hunk 8]
```diff
@@ -1,14 +1,12 @@
 import React, { Component } from 'react';
```
Comment (fzaninotto) [inline src/app/js/formats/resources-grid/Component.js:1]: rename to `ResourcesGrid`

File: src/app/js/formats/resources-grid/Component.js [hunk 9]
```diff
@@ -36,30 +33,26 @@ class ResourcesGrid extends Component {
...
-        };
+        const queryString = getQueryString({
+            sort,
+            params: { maxSize: maxSize + this.state.more },
+        });
 
-        const apiurl = url.format(url.format(forJSON));
+        const url = `${resource[field.name]}?${queryString}`;
+
+        const apiurl = url;
```
Comment (fzaninotto) [inline src/app/js/formats/resources-grid/Component.js]: code smell

File: src/app/js/public/graph/GraphItem.js [hunk 10]
```diff
@@ -1,25 +1,58 @@
...
+import { fromCharacteristic, fromGraph } from '../selectors';
 import GraphItemContainer from './GraphItemContainer';
+import { preLoadChartData } from '../graph';
 
-const PureGraphItem = ({ field, resource }) => (
-    <GraphItemContainer link={`/graph/${field.name}`} label={field.label}>
-        <Format field={field} resource={resource} />
-    </GraphItemContainer>
-);
+class PureGraphItem extends Component {
```
Comment (fzaninotto) [inline src/app/js/public/graph/GraphItem.js]: should not be `Pure`

File: src/app/js/public/graph/GraphPage.js [hunk 11]
```diff
@@ -39,63 +45,82 @@ const styles = {
...
-            <Card style={styles.section}>
-                <Dataset />
-            </Card>
-        </div>
-        <div style={styles.sideColumn}>
-            <Toolbar />
-        </div>
-    </div>
-);
+class PureGraphPage extends Component {
```
Comment (fzaninotto) [inline src/app/js/public/graph/GraphPage.js]: should not be `Pure`

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: src/api/statements/LodexReduceQuery.js (hunk 1 above)
Comment (ThieryMichel) [reply src/api/statements/LodexReduceQuery.js:38]: ezs does not allow to do that.

File: src/app/js/formats/resources-grid/Component.js (hunk 8 above)
Comment (ThieryMichel) [reply src/app/js/formats/resources-grid/Component.js:1]: done

File: src/api/statements/LodexReduceQuery.js (hunk 1 above)
Comment (parmentf) [reply src/api/statements/LodexReduceQuery.js:38]: I think it's possible with ezs, using the configuration object, and `getParam`.

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**